### DISCUSSION TITLE
Is there a way to get a select widget to "drop down" automatically?

### DISCUSSION THREAD

**Discussion by bob-anderson-ok** [fyne-io/fyne#4667] [Q&A]
I have a select widget inside a popup window. I would like the select widget to "drop down/open the list" automatically. That way the user does not have to click (every time) on the widget just to see the select list.

**Comment by andydotxyz** [comment]
You can just call `Tapped` on the widget I would think - that will make it think that it was clicked / tapped and therefore open.

**Comment by bob-anderson-ok** [comment]
Tapped() needs a parmeter (*fyne.PointEvent)

I can't get past that requirement. I tried this ...

fakedPoint := fyne.PointEvent{
		AbsolutePosition: fyne.Position{X: 0, Y: 0},
		Position:         fyne.Position{X: 0, Y: 0},
	}
item1.Tapped(&fakedPoint)

but that causes a panic.

**Comment by andydotxyz** [comment]
The Position is how far into the widget you want to tap (on a select that won't matter, as it's all one tappable area).
The absolute may be required for the widget to calculate the position, I cannot recall - but the driver has an "AbsolutePositionForWidget" which you can use to add the position to for an absolute position to simulate.

**Comment by bob-anderson-ok** [comment]
I got this to work. The delay in openSelections() is rather hacky, but it was needed.

I include this for anyone following this thread with a similar problem in spite of the 'hack'.

```
func folderHistorySelect() {
	selector := widget.NewSelect(myWin.fitsFolderHistory, func(path string) { processFolderSelection(path) })
	myWin.folderSelect = selector // Save for use by openSelections()

	// Configure selector
	selector.PlaceHolder = "Folder history"
	folderSelectWin := myWin.App.NewWindow("FITS folder history (and options)")
	myWin.folderSelectWin = folderSelectWin
	folderSelectWin.Resize(fyne.Size{Height: 450, Width: 700})

	ctr := container.NewVBox()
	ctr.Add(selector)
	ctr.Add(layout.NewSpacer())
	folderSelectWin.SetContent(ctr)
	folderSelectWin.CenterOnScreen()

	folderSelectWin.SetCloseIntercept(func() { processFolderSelection("") })
	folderSelectWin.Show()

	go openSelections()
}

func openSelections() {
	// The following delay is apparently needed to allow the 
	// window with its widgets to be built, particularly the Select widget
	time.Sleep(100 * time.Millisecond)  
	fakedPoint := fyne.PointEvent{
		AbsolutePosition: myWin.folderSelect.Position(),
		Position:         myWin.folderSelect.Position(),
	}
	myWin.folderSelect.Tapped(&fakedPoint)
}
```
